targets/TARGET_ARM_SSG/TARGET_MUSCA_A1/device/COMPONENT_NSPE/TOOLCHAIN_ARMC6/musca_ns.sct

```diff
@@ -0,0 +1,37 @@
+#! armclang --target=arm-arm-none-eabi -march=armv8-m.main -E -xc
+
+/*
+ * Copyright (c) 2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "../cmsis_nvic.h"
+
+LR_CODE NS_CODE_START NS_CODE_SIZE {
```

Comment:
`NS_CODE_START` and `NS_CODE_SIZE` - Where are these defines set? Are those configurable?

Reply:
They are defined in [region_defs.h](https://github.com/ARMmbed/mbed-os/blob/b44fb46c39a831ac94433cfa0b66198ca740f245/targets/TARGET_ARM_SSG/TARGET_MUSCA_A1/partition/region_defs.h), _cmsis_nvic.h_ includes this file.
But to be honest I think I will add an explicit include in the ld/sct files as well (explicit is better than implicit 😄)

Reply:
Done

Reply:
@mikisch81 - We have few common defines (RAM / ROM start and size) already available in the tools, which are picked from CMSIS packages and passed as defines. It will be good to use the same instead of adding new ones. 

Please check in build of MUSCA if `MBED_RAM_START` `MBED_ROM_START` are defined, if yes you can use them

Reply:
@deepikabhavnani Yes, I know these defines, we use them in PSOC6.
MUSCA build doesn't define them, it takes the values from `region_defs.h` which is from the TF-M port of Musca.
I can add a block like in [PSOC6](https://github.com/ARMmbed/mbed-os/blob/master/targets/TARGET_Cypress/TARGET_PSOC6/TARGET_CY8C63XX/TARGET_MCU_PSOC6_M4/device/TOOLCHAIN_GCC_ARM/cy8c6xx7_cm4_dual.ld#L31-L45) which will set these defines accordingly.

Reply:
Will appreciate if defines in `region_defs.h` and linker files are updated and used as `PSOC6`, it will give flexibility to users instead of hard-coded values.

Reply:
I think it should be relevant only to NS side, as the secure side only runs TF-M on bare-metal.
Something like this in the NS linker scripts:
```c
#if !defined(MBED_ROM_START)
  #define MBED_ROM_START    NS_CODE_START
#endif

#if !defined(MBED_ROM_SIZE)
  #define MBED_ROM_SIZE     ((TOTAL_CODE_SRAM_SIZE / 2) - BL2_HEADER_SIZE)
#endif

#if !defined(MBED_RAM_START)
  #define MBED_RAM_START    NVIC_RAM_VECTOR_LIMIT
#endif

#if !defined(MBED_RAM_SIZE)
  #define MBED_RAM_SIZE     (NS_DATA_SIZE - NVIC_RAM_VECTOR_SIZE)
#endif
```

Reply:
TF-M though a bare-metal layer, it has linker files https://github.com/ARMmbed/mbed-os/pull/9221/files#diff-477bb268cd4ef63fbe0f6f161bbfeb7cR25

Change is relevant for secure + non-secure side as well. Idea is to have defines at one place, instead of hard-coded in code, so if someone wants to change the size of secure / non-secure RAM / ROM it all should be at one place. Now it looks like it is `region_defs.h`  

I will propose something like this instead

```
#if !defined(MBED_ROM_START)
  #define MBED_ROM_START    0x10200000
#endif

#if !defined(MBED_ROM1_START)
  #define MBED_ROM1_START    0x00200000
#endif

#define S_ROM_ALIAS_BASE           MBED_ROM_START
#define NS_ROM_ALIAS_BASE        MBED_ROM1_START
#define TOTAL_ROM_SIZE      (MBED_ROM_SIZE + MBED_ROM1_SIZE)
```

MBED_ROM_START / MBED_ROM_SIZE - Will map to IROM1
MBED_ROM1_START / MBED_ROM1_SIZE - Will map to IROM2
![image](https://user-images.githubusercontent.com/28542450/50661682-788d0000-0f69-11e9-913b-3c46dad471db.png)